source/CIPlanetNode.cpp

```diff
@@ -33,19 +33,31 @@ void PlanetNode::update(float timestep) {
     for (int ii = 0; ii < _layers->size(); ii++) {
       LayerNode* node = &_layerNodes[ii];
       if (node != nullptr && node->innerRing) {
-        advanceFrame(node);
+        bool isLockedIn = _layers->at(ii).isLockedIn;
+        advanceInnerLayerFrame(node);
+        advanceOuterLayerFrame(node, isLockedIn, isLockingIn, ii, numLayers);
       }
     }
   }
 }
 
-void PlanetNode::advanceFrame(LayerNode* node) {
-  // Our animation depends on the current frame
-  unsigned int frame = node->innerRing->getFrame();
-  frame = (frame == INNER_RING_END) ? INNER_RING_START : frame + 1;
-  node->innerRing->setFrame(frame);
+void PlanetNode::advanceInnerLayerFrame(LayerNode* node){
+  unsigned int innerFrame = node->innerRing->getFrame();
+  innerFrame = (innerFrame == INNER_RING_END) ? INNER_RING_START : innerFrame + 1;
+  node->innerRing->setFrame(innerFrame);
 }
 
+void PlanetNode::advanceOuterLayerFrame(LayerNode* node, bool isLockedIn, bool isLockingIn, int ii, int numLayers){
```

Comment:
NIT: instead of passing in ii, you could just set isLockingIn to true only for the outermost layer